parser/src/main/java/io/specmesh/apiparser/model/Message.java

```diff
@@ -40,12 +42,17 @@
 @SuppressFBWarnings
 @SuppressWarnings({"unchecked", "rawtypes"})
 public class Message {
+    static final Set<String> SUPPORTED_TYPES =
+            Set.of("UUID", "long", "int", "short", "float", "double", "string", "bytes");
+
     @JsonProperty private String messageId;
 
     @JsonProperty private Map headers = Collections.EMPTY_MAP;
 
     @JsonProperty private Map payload = Collections.EMPTY_MAP;
 
+    @JsonProperty private Map key = Collections.EMPTY_MAP;
```

Comment:
Rather than an un-typed `Map`, have you considered having a specific PoJo to model the exact fields SpecMesh supports for `key`?

e.g. 

```java
public final class PayloadProperties {
    public PayloadPart key();
    public PayloadPart value();
}

public final class PayloadPart {
    public TypeInfo type();
}
```

Reply:
not relevant